crates/storage/provider/src/providers/mod.rs

```diff
@@ -105,40 +105,29 @@ where
         self.database.best_block_number()
     }
 
-    fn convert_block_number(&self, num: BlockNumberOrTag) -> Result<Option<BlockNumber>> {
-        let num = match num {
-            BlockNumberOrTag::Latest => self.chain_info()?.best_number,
-            BlockNumberOrTag::Number(num) => num,
-            BlockNumberOrTag::Pending => {
-                return Ok(self.tree.pending_block_num_hash().map(|b| b.number))
-            }
-            BlockNumberOrTag::Finalized => return Ok(self.chain_info()?.last_finalized),
-            BlockNumberOrTag::Safe => return Ok(self.chain_info()?.safe_finalized),
-            BlockNumberOrTag::Earliest => 0,
-        };
-        Ok(Some(num))
-    }
-
-    fn block_hash_for_id(&self, block_id: BlockId) -> Result<Option<H256>> {
-        match block_id {
-            BlockId::Hash(hash) => Ok(Some(hash.into())),
-            BlockId::Number(num) => match num {
-                BlockNumberOrTag::Latest => Ok(Some(self.chain_info()?.best_hash)),
-                BlockNumberOrTag::Pending => Ok(self.tree.pending_block_num_hash().map(|b| b.hash)),
-                _ => self
-                    .convert_block_number(num)?
-                    .map(|num| self.block_hash(num))
-                    .transpose()
-                    .map(|maybe_hash| maybe_hash.flatten()),
-            },
-        }
-    }
-
     fn block_number(&self, hash: H256) -> Result<Option<BlockNumber>> {
         self.database.block_number(hash)
     }
 }
 
+impl<DB, Tree> BlockIdProvider for BlockchainProvider<DB, Tree>
+where
+    DB: Database,
+    Tree: BlockchainTreeViewer + Send + Sync,
+{
+    fn safe_block_num(&self) -> Result<Option<reth_primitives::BlockNumber>> {
+        todo!()
+    }
+
+    fn finalized_block_num(&self) -> Result<Option<reth_primitives::BlockNumber>> {
+        todo!()
+    }
```

Comment:
Is this never getting hit? Or should it return an error?

Reply:
Pushed this #2628 until we merge the tracker, which should handle it completely